File: Daily Challanges/2026-07-July/2026-07-01.js
function getLuckyNumber(name) {
  const [first, last] = name.split(" ");

  function counts(s) {
    let vowels = 0;
    let consonants = 0;

    for (const c of s.toLowerCase()) {
      if ("aeiou".includes(c)) {
        vowels++;
      } else if (/[a-z]/.test(c)) {
        consonants++;
      }
    }

    return [vowels, consonants, s.length];
  }

  const [v1, c1, l1] = counts(first);
  const [v2, c2, l2] = counts(last);

  const small = Math.min(v1, v2) * Math.min(c1, c2) * Math.min(l1, l2);
  const large = Math.max(v1, v2) * Math.max(c1, c2) * Math.max(l1, l2);

  const ans = large - small;
  return ans === 0 ? 13 : ans;
}

console.log(getLuckyNumber("John Doe"));
console.log(getLuckyNumber("Olivia Lewis"));
console.log(getLuckyNumber("James Wilson"));
console.log(getLuckyNumber("Elizabeth Hernandez"));
console.log(getLuckyNumber("Mike Walker"));
console.log(getLuckyNumber("Chloe Perez"));

File: Daily Challanges/2026-07-July/2026-07-01.py
def get_lucky_number(name):
    first, last = name.split()
    def counts(s):
        vowels = sum(c.lower() in "aeiou" for c in s)
        consonants = sum(c.isalpha() and c.lower() not in "aeiou" for c in s)
        return vowels, consonants, len(s)

    v1, c1, l1 = counts(first)
    v2, c2, l2 = counts(last)

    small = min(v1, v2) * min(c1, c2) * min(l1, l2)
    large = max(v1, v2) * max(c1, c2) * max(l1, l2)

    ans = large - small
    return 13 if ans == 0 else ans


print(get_lucky_number("John Doe"))
print(get_lucky_number("Olivia Lewis"))
print(get_lucky_number("James Wilson"))
print(get_lucky_number("Elizabeth Hernandez"))
print(get_lucky_number("Mike Walker"))
print(get_lucky_number("Chloe Perez"))